lib/global.h

```diff
@@ -9,11 +9,15 @@
 
 #include <glib.h>
 
-#if defined(HAVE_FUNC_ATTRIBUTE_WEAK) && defined(HAVE_TESTS)
+#if defined __has_attribute
```

Comment:
you're leaving the various constants undefined when this is undefined.

you should probably define an MC_HAS_ATTRIBUTE macro for that purpose.

Reply:
Thanks for noticing, I really hate that macro stuff. I hope it's fixed now.